backend/api/Areas/Documents/DocumentController.cs

```diff
@@ -38,152 +37,135 @@ public class DocumentController : ControllerBase
+        /// <returns></returns>
+        [HttpGet("{relationshipType}/{parentId:long}")]
         [Produces("application/json")]
-        [ProducesResponseType(typeof(ExternalResult<QueryResult<DocumentDetail>>), 200)]
-        [SwaggerOperation(Tags = new[] { "documents" })]
-        public IActionResult GetDocumentList()
+        //[HasPermission(Permissions.NoteView)]
+        [ProducesResponseType(typeof(IList<DocumentRelationshipModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "document" })]
+        public IActionResult GetActivityDocuments(DocumentRelationType relationshipType, long parentId)
         {
-            var result = _documentService.GetStorageDocumentList();
-            return new JsonResult(result);
+            switch (relationshipType)
+            {
+                case DocumentRelationType.Activities:
+                    var documents = _documentService.GetActivityDocuments(parentId);
+                    var mapedDocuments = _mapper.Map<List<DocumentRelationshipModel>>(documents);
+                    return new JsonResult(mapedDocuments);
+                default:
+                    throw new BadRequestException("Relationship type not valid.");
+            }
         }
 
         /// <summary>
         /// Downloads the file for the corresponding file and document id.
         /// </summary>
-        [HttpGet("{documentId}/files/{fileId}/download")]
+        [HttpGet("storage/{mayanDocumentId}/files/{mayanFileId}/download")]
         //[HasPermission(Permissions.PropertyAdd)]
         [ProducesResponseType(typeof(ExternalResult<FileDownload>), 200)]
-        [SwaggerOperation(Tags = new[] { "documents" })]
-        public async Task<IActionResult> DownloadFile(int documentId, int fileId)
+        [SwaggerOperation(Tags = new[] { "storage-documents" })]
+        public async Task<IActionResult> DownloadFile(long mayanDocumentId, long mayanFileId)
         {
-            var result = await _documentService.DownloadFileAsync(documentId, fileId);
+            var result = await _documentService.DownloadFileAsync(mayanDocumentId, mayanFileId);
             return new JsonResult(result);
         }
 
         /// <summary>
-        /// Downloads the latest file for the corresponding document id.
+        /// Retrieves a list of documents.
         /// </summary>
-        [HttpGet("{documentId}/download")]
-        //[HasPermission(Permissions.PropertyAdd)]
-        [ProducesResponseType(typeof(ExternalResult<FileDownload>), 200)]
-        [SwaggerOperation(Tags = new[] { "documents" })]
-        public async Task<IActionResult> DownloadFile(int documentId)
+        [HttpGet("storage")]
+        [HasPermission(Permissions.PropertyAdd)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ExternalResult<QueryResult<DocumentDetail>>), 200)]
+        [SwaggerOperation(Tags = new[] { "storage-documents" })]
+        public IActionResult GetDocumentList()
```

Comment:
So this is just all the documents in the system then? I guess just for the test page?

Reply:
Yes, this gets all documents. If it turns out that we dont needed I will remove it